README.md

```diff
@@ -1,61 +1,2577 @@
+636c8f11fee2        simple-docker-app   "/usr/bin/entry.sh /…"   3 minutes ago       Up 3 minutes        0.0.0.0:32769->9854/tcp   epic_montalcini
+```
+Interestingly, whilst port 9854 in the container *is* mapped a port on the host,
+it's not mapped to port 9854. This is because `--publish-all` maps ports
+specified by the `EXPOSE` directive to
+[ephemeral ports](https://en.wikipedia.org/wiki/Ephemeral_port) on the host,
+usually the next free one. This initially doesn't sound particularly useful,
+but becomes more so in the context of Docker deployments to Cloud hosts where
+services are usually run.
+
+If we now try `curl` again in T2 using port 32769 (this will most likely change
+on your own machine, ensure you use the right mapped port), we'll see that all
+works as expected:
+```
+$ curl -XPOST http://localhost:32769/echo -H 'Content-type: text/plain' -d 'Hello there!'
+Hello there!
+```
+Any Docker image which has been built with the `EXPOSE` directive can still be
+forced to map a specific host port to a specific container port just as before
+by using the `--publish` switch to `docker run`.
+
+Whilst the use of `EXPOSE` is not mandatory when writing Dockerfiles, its use
+is encouraged as useful documentation and metadata to those who want to run your
+Docker images. This stops extra documentation being required to state explicitly
+which ports your image uses, and also means when you update your image, all a
+user has to do is see if extra `EXPOSE` directives have been used and then map
+the new ports appropriately to the host.
+
+Finally, stop the container running in T1 and then close T2. We'll refer to
+T1 from this point as 'the terminal' again.
+
+#### 6.6 Container Instances
+
+One thing you may have spotted whilst we were stopping and rerunning our Docker
+image was that the name and the ID of the container kept changing.
+
+In the terminal, run the following:
+```
+$ docker ps --all
+CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS                         PORTS               NAMES
+636c8f11fee2        simple-docker-app   "/usr/bin/entry.sh /…"   20 minutes ago      Exited (0) 3 minutes ago                           epic_montalcini
+0579294093b6        f50ceee6da24        "/usr/bin/entry.sh /…"   51 minutes ago      Exited (0) 26 minutes ago                          jolly_banzai
+61578a11c16e        d00c59983bd8        "/usr/bin/entry.sh /…"   About an hour ago   Exited (0) About an hour ago                       tender_gagarin
+```
+However, looking at the list, this seems strange at first. We we just kept
+running our `simple-docker-app`, so why are there several instances of it? Well,
+every time we used `docker run simple-docker-app`, we were asking the Docker
+daemon to run a *new* instance of the image, not to run the same container
+image that we had before.
```

Comment:
Worth adding another quick reference to docker start for the case to reuse an existing stopped container?

Reply:
Not adding another example, but I've reference the command.